src/lib/search-params.ts

```diff
@@ -29,49 +29,47 @@ const sortMapping = {
   },
 };
 
+const searchParamArraySchema = z
+  .array(z.string())
+  .transform(stringArray => stringArray.join(','))
+  .optional();
+
+const searchParamsSchema = z.object({
+  query: z.string().optional(),
+  offset: z
+    .number()
+    // Don't add the default offset of 0 to the search params
+    .transform(offset => (offset > 0 ? `${offset}` : undefined))
+    .optional(),
+  licenses: searchParamArraySchema,
+  publishers: searchParamArraySchema,
+  spatialCoverages: searchParamArraySchema,
+  sortBy: z
+    .nativeEnum(SortBy)
+    // Don't add the default sort to the search params
+    .transform(sortBy => (sortBy === defaultSortBy ? undefined : sortBy))
+    .optional(),
+});
+
 interface ClientSearchOptions {
   query?: SearchOptions['query'];
   offset?: SearchOptions['offset'];
-  filters: {
-    licenses?: string[];
-    publishers?: string[];
-    spatialCoverages?: string[];
-  };
+  licenses?: string[];
+  publishers?: string[];
+  spatialCoverages?: string[];
   sortBy?: SortBy;
 }
 
-export function getUrlWithSearchParams({
-  query,
-  offset,
-  filters: {licenses, publishers, spatialCoverages},
-  sortBy,
-}: ClientSearchOptions): string {
-  const searchParams: {[key: string]: string} = {};
-
-  if (query) {
-    searchParams.query = query;
-  }
-
-  if (licenses?.length) {
-    searchParams.licenses = licenses.join(',');
-  }
-
-  if (publishers?.length) {
-    searchParams.publishers = publishers.join(',');
-  }
-
-  if (spatialCoverages?.length) {
-    searchParams.spatialCoverages = spatialCoverages.join(',');
-  }
-
-  if (offset) {
-    searchParams.offset = `${offset}`;
-  }
-
-  if (sortBy && sortBy !== defaultSortBy) {
-    searchParams.sortBy = sortBy;
-  }
+export function getUrlWithSearchParams(
+  clientSearchOption: ClientSearchOptions
+): string {
+  const searchParams: {[key: string]: string} =
+    searchParamsSchema.parse(clientSearchOption);
 
+  // Only add relevant values to the search params. Remove all keys with a empty value (undefined or "")
+  Object.keys(searchParams).forEach(key =>
+    !searchParams[key] ? delete searchParams[key] : {}
```

Comment:
Very minor: this deletes all falsy values (not just undefined and empty string, but also `0` and `null`)?

Reply:
Since all values should be an optional string after the parse, I figured this was safe.

Reply:
👍🏻